### PR TITLE
User

### PR DESCRIPTION
**Pull request by BrunoLagasse**

### CODE REVIEW

File: src/Controller/HomeController.php [hunk 1]
```diff
@@ -17,11 +18,20 @@ public function doormat(): Response
...
+        // if ($this->getUser()) {
+        //     return $this->redirectToRoute('target_path');
+        // }
+
+        // get the login error if there is one
+        $error = $authenticationUtils->getLastAuthenticationError();
+        // last username entered by the user
+        $lastUsername = $authenticationUtils->getLastUsername();
+
+        return $this->render('security/login.html.twig', ['last_username' => $lastUsername, 'error' => $error]);
```
Comment (rocambille) [inline src/Controller/HomeController.php:34]: you declared this route twice: see SecurityController

File: src/Security/LoginFormAuthenticator.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $providerKey)
+    {
+        if ($targetPath = $this->getTargetPath($request->getSession(), $providerKey)) {
+            return new RedirectResponse($targetPath);
+        }
+
+        // For example : return new RedirectResponse($this->urlGenerator->generate('some_route'));
+        throw new \Exception('TODO: provide a valid redirect inside '.__FILE__);
```
Comment (rocambille) [inline src/Security/LoginFormAuthenticator.php:100]: missing redirection here

File: templates/_header.html.twig [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {% block header %}
     <header>
         <a href="{{ app.request.headers.get('referer') }}"><img class="leftarrow" src="{{ asset('build/images/previous.png') }}" alt="leftarrow"/></a>
-        <a href=""><img class="user-icon" src="{{ asset('build/images/usericon.png') }}" alt="user-icon"/></a>
+        <a href="/user/4"><img class="user-icon" src="{{ asset('build/images/usericon.png') }}" alt="user-icon"/></a>
```
Comment (rocambille) [inline templates/_header.html.twig:4]: use path() to generate urls

**Review by rocambille — CHANGES_REQUESTED** [review]

**Comment by rocambille** [comment]
you shouldn't have deleted these files:

* .github/workflows/deploy.yml
* .github/workflows/main.yml

restore them

**Comment by rocambille** [comment]
and conflicts to resolve ;)

**Comment by BrunoLagasse** [comment]
Delete this pull request

**Pull request closed without merging**